samples/ipc/ipc_service/src/main.c

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr.h>
+#include <device.h>
+
+#include <ipc/ipc_service.h>
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(host, LOG_LEVEL_INF);
+
+#define MLEN_0 40
+
+struct data_packet {
+	unsigned char message;
+	unsigned char data[100];
+};
+
+
+K_SEM_DEFINE(bound_sem, 0, 1);
+
+static void ep_bound(void *priv)
+{
+	k_sem_give(&bound_sem);
+}
+
+static void ep_recv(const void *data, size_t len, void *priv)
+{
+	char received_data = *((char *)data);
+	static char expected_message = 'a';
+	static uint16_t expected_len = MLEN_0;
+
+	static unsigned long long cnt;
+	static unsigned int stats_every;
+	static uint32_t start;
+	static unsigned int err;
+
+	if (start == 0) {
+		start = k_uptime_get_32();
+	}
+
+	if (received_data != expected_message) {
+		err++;
+	}
+
+	if (len != expected_len) {
+		err++;
+	}
+
+	expected_message++;
+	expected_len++;
+
+	cnt += len;
+
+	if (expected_message > 'z') {
+		if (stats_every++ > 50) {
+			/* Print throuhput [Bytes/s]. Use printk not to overload CPU with logger.
+			 * Sample never reaches lower priority thread because of high throughput
+			 * (100% cpu load) so logging would not be able to handle messages in
+			 * deferred mode (immediate mode would be heavier than printk).
+			 */
+			printk("%llu\n", (1000*cnt)/(k_uptime_get_32() - start));
+			stats_every = 0;
+		}
+		if (err) {
+			printk("Unexpected message\n");
+		}
+		expected_message = 'a';
+	}
+
+	if (expected_len > sizeof(struct data_packet)) {
+		expected_len = MLEN_0;
```

Comment:
I get it - we are switching message len :)

can we change MLEN_0 to BASE_MESSAGE_LEN?
and use variable len_change (i.e. expected_len = BASE_MESSAGE_LEN + len_change) instread of expected_len

Reply:
would be self documenting